src/base/memory_monitor_sr.cpp

```diff
@@ -0,0 +1,235 @@
+/*
+ *
+ * Copyright 2016 CUBRID Corporation
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+/*
+ * memory_monitor_sr.cpp - implementation of memory monitoring manager
+ */
+
+#include <stdexcept>
+#include <cstring>
+
+#include "perf.hpp"
+#include "error_manager.h"
+#include "memory_monitor_sr.hpp"
+
+#include "log_impl.h"
+
+namespace cubperf
+{
+  memory_monitor *mmon_Gl = nullptr;
+
+  const char *mmon_subcomponent::get_name ()
+  {
+    return m_subcomp_name.c_str ();
+  }
+
+  void mmon_subcomponent::add_cur_stat (uint64_t size)
+  {
+    m_cur_stat.fetch_add (size);
+  }
+
+  void mmon_subcomponent::sub_cur_stat (uint64_t size)
+  {
+    m_cur_stat.fetch_sub (size);
+  }
+
+  const char *mmon_component::get_name ()
+  {
+    return m_comp_name.c_str ();
+  }
+
+  void mmon_component::add_stat (uint64_t size, int subcomp_idx, bool init, bool expand)
+  {
+    /* description of add_stat(..).
+     * 1) if init == true, add init_stat
+     * 2) then, add cur_stat
+     * 3) compare with peak_stat
+     * 4) if cur_stat(new) > peak_stat, update peak_stat
+     * 5) if expand == true, expand_count++
+     * 6) call subcomponent->add_cur_stat(size) */
+    return;
+  }
+
+  void mmon_component::sub_stat (uint64_t size, int subcomp_idx, bool init)
+  {
+    /* description of sub_stat(size, init).
+     * 1) if init == true, sub init_stat
+     * 2) then, sub cur_stat
+     * 3) call component->sub_stat(size, subcomp_idx, init) */
+    return;
+  }
+
+  int mmon_component::is_subcomp_exist (const char *subcomp_name)
```

Comment:
```suggestion
  int mmon_component::get_subcomp_index (const char *subcomp_name)
```
seems more correct based on the function behavior.

Reply:
You're right. I change it.